feat(billing): add Billing Summary Card (#116)

sapatmohit opened the pull request:
## Summary
This PR introduces the **Billing Summary Card** component as part of the Billing and Usage Components initiative (#116).  
It displays plan details, renewal date, and provides an upgrade action.

Closes #116 
Closes #55 

## Changes
- Added **Billing Summary Card** UI with plan and renewal date.  
- Included **Upgrade Button** for plan upgrades.  
- Ensured responsiveness, accessibility, and consistent styling.  

## Screenshots/Recordings (UI)
Billing Summary Card is shown in both themes:

**Light Theme**  
<img width="1121" height="476" alt="Billing Summary Card Light Theme" src="https://github.com/user-attachments/assets/0df2871c-20a9-4f27-99e6-95fdc813f61c" />

**Dark Theme**  
<img width="1125" height="480" alt="Billing Summary Card Dark Them" src="https://github.com/user-attachments/assets/96157908-7ae4-4c1a-8a8e-3230e3481f2e" />

## How to Test
1. Checkout this branch.  
2. Run `npm ci`.  
3. Run `npm run typecheck` and `npm run build`.  
4. Start the app and verify that the **Billing Summary Card** renders correctly in both light and dark themes.

## Checklist
- [x] Title is clear and descriptive  
- [x] Linked to related issue (#116)  
- [x] Tests or manual verification steps included  
- [x] CI passes (typecheck & build)  
- [x] Docs updated (if needed)  


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Added a Billing Summary Card component with plan details and an Upgrade action, plus interactive demos.

* **Documentation**
  * New Billing Summary Card docs page with installation paths (shadcn & billingSDK), usage examples, props, and credits.
  * Added the page to navigation under Billing & Usage Analytics; updated components index and Billing Setting copy.

* **Style**
  * Reformatted component listings into a padded two-column layout for improved readability.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

tsahil01 commented on content/docs/components/index.mdx:
@sapatmohit i think by mistake you changed this files formating. please revert this file and then add your component

tsahil01 commented on src/components/billing-summary-card-demo.tsx:
Here no need to export it, Just copy the demo file from registry and paste it here.

tsahil01 reviewed (changes requested):
Rest is cool,
Nice PR. Fix those changes, then will merge it.

tsahil01 commented:
You need to fix those changes in all of your PRs.

sapatmohit commented:
> You need to fix those changes in all of your PRs.

@tsahil01 I’ve fixed the changes from my side. Could you review and let me know if everything looks correct? If it’s good, I’ll apply the same fixes across my other PRs.

tsahil01 reviewed (changes requested):
Well this PR is almost ready for merge, only small fixes. 
Please change these things in other PRs as well.
I just wanted to make sure every other PR follows the same approach, will make merging easy for us.
Thanks.

sapatmohit commented:
> Well this PR is almost ready for merge, only small fixes. Please change these things in other PRs as well. I just wanted to make sure every other PR follows the same approach, will make merging easy for us. Thanks.

@tsahil01 Sorry about that! I’ve corrected it and also updated all my other components accordingly.
